src/core/server/models/story/index.ts

```diff
@@ -501,3 +521,137 @@ export async function updateStoryLastCommentedAt(
     }
   );
 }
+
+export async function addExpert(
+  mongo: Db,
+  tenantID: string,
+  storyID: string,
+  userID: string
+) {
+  const story = await collection(mongo).findOne({ tenantID, id: storyID });
+  if (!story) {
+    throw new StoryNotFoundError(storyID);
+  }
+  if (
+    story.settings.expertIDs &&
+    story.settings.expertIDs.some(e => e === userID)
+  ) {
+    return story;
+  }
```

Comment:
I disagree, this is important to ensure that we don't add a duplicate expert ID onto the story.

I don't want safety checks spread across a service layer was well as the model (repository). This is a data specific problem, the service does not care about duplicates, it cares about the adding/removing. These checks I believe belong in the data layer.

Reply:
So, given that the below database code is using a `findOneAndUpdate` method, it actually performs the find operation for you, for free without any additional database access being required. The use of the `$addToSet` operation also actually prevents duplicates from being added to the set, thereby ensuring that the userID is not duplicated.

I would check to see the remainder of the function additions in this file do not query the database unnecessarily for these operations. You can instead just do:

```ts
if (!result.value) {
  throw new StoryNotFoundError(storyID);
}
```

Which can accomplish the same check without any additional database access.

The idea/drive behind this suggestion is that we want to ensure that we use atomic update operations when making document updates because we can then remove the pre-checks before these updates are performed.

I would also ensure that new functions added to this file include the name `Story` in the function name to stay consistent with the existing function naming.